From 5b7202be201b4798c2f237d84ad0a2936b104d05 Mon Sep 17 00:00:00 2001
From: Konrad Kujawa <konrad.kujawa@qt.io>
Date: Thu, 29 Jun 2023 13:33:09 +0200
Subject: [PATCH] Remove call to protobuf FileDescriptor::syntax()

In the new version of protobuf FileDescriptor::syntax() should not
be used. Instead, protobuf API should be working with both,
proto2 and proto3 syntax.

Task-number: QTBUG-114833
Task-number: QTBUG-114962
Change-Id: I86542cfcc0cc247f1e75e6d711e269ff3537f8a4
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
(cherry picked from commit 2297f2da520b7eebe10dd6b5304e5a4c3c18a57c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
Upstream-Status: Backport [6.6.0 e55cc9d Remove call to protobuf FileDescriptor::syntax()]

 src/tools/qtgrpcgen/qgrpcgenerator.cpp         | 6 +-----
 src/tools/qtprotobufgen/qprotobufgenerator.cpp | 7 +------
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/src/tools/qtgrpcgen/qgrpcgenerator.cpp b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
index ef2ce62..62c233d 100644
--- a/src/tools/qtgrpcgen/qgrpcgenerator.cpp
+++ b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
@@ -30,13 +30,9 @@ QGrpcGenerator::~QGrpcGenerator() = default;
 bool QGrpcGenerator::Generate(const FileDescriptor *file,
                               [[maybe_unused]] const std::string &parameter,
                               GeneratorContext *generatorContext,
-                              std::string *error) const
+                              [[maybe_unused]] std::string *error) const
 {
     assert(file != nullptr && generatorContext != nullptr);
-    if (file->syntax() != FileDescriptor::SYNTAX_PROTO3) {
-        *error = "Invalid proto used. qtgrpcgen only supports 'proto3' syntax";
-        return false;
-    }
 
     return GenerateClientServices(file, generatorContext);
 }
diff --git a/src/tools/qtprotobufgen/qprotobufgenerator.cpp b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
index 941ef97..e46864d 100644
--- a/src/tools/qtprotobufgen/qprotobufgenerator.cpp
+++ b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
@@ -35,15 +35,10 @@ QProtobufGenerator::~QProtobufGenerator() = default;
 bool QProtobufGenerator::Generate(const FileDescriptor *file,
                                   [[maybe_unused]] const std::string &parameter,
                                   GeneratorContext *generatorContext,
-                                  std::string *error) const
+                                  [[maybe_unused]] std::string *error) const
 {
     assert(file != nullptr && generatorContext != nullptr);
 
-    if (file->syntax() != FileDescriptor::SYNTAX_PROTO3) {
-        *error = "Invalid proto used. qtprotobufgen only supports 'proto3' syntax";
-        return false;
-    }
-
     return GenerateMessages(file, generatorContext);
 }
 
